Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -268,43 +283,55 @@ void LeptonTrackingSFProducer::Init(ZJetSettings const& settings)
         m_ptbins = &m_ybins;
     }
     // Get file
-    LOG(INFO) << "Loading lepton scale factors for Tracking: File " << m_sffile
+    LOG(INFO) << "Loading lepton scale factors for " << m_id << ": File " << m_sffile
               << ", Histogram " << histoname;
     TFile file(m_sffile.c_str(), "READONLY");
-    TGraphAsymmErrors *sfhisto = (TGraphAsymmErrors*)file.Get(histoname.c_str());
-
+    TH2F* sfhisto = (TH2F*)file.Get(histoname.c_str());
+    if (!m_2D)
+        // This step causes an unused variable warning. How can we get rid of it?
+        TH1F* sfhisto = (TH1F*)file.Get(histoname.c_str());
+    
     // Get the pT and eta bin borders
-    m_xbins.emplace_back(sfhisto->GetX()[0] - sfhisto->GetErrorXlow(0));
-    for (int ix = 0; ix < sfhisto->GetN(); ++ix){
-        m_xbins.emplace_back(sfhisto->GetX()[ix] + sfhisto->GetErrorXhigh(ix));
+    for (int ix = 0; ix <= sfhisto->GetNbinsX(); ++ix){
+        m_xbins.emplace_back(2 * sfhisto->GetXaxis()->GetBinCenter(ix) -
+                             sfhisto->GetXaxis()->GetBinLowEdge(ix));
```

Comment:
Gibt es für die Änderung einen speziellen Grund? Soweit ich das verstehe, sollten beide Wege zum selben Ergebnis führen.

Reply:
Tatsächlich nicht. Ich habe den Code von den anderen LeptonSF (Iso und ID) kopiert und da stand das so drin.

Reply:
Ich glaube das war der Teil der bei mir beim testen einen Segfault hervorgerufen hat. Mit dem Code wie in den anderen LeptonSFs hats dann getan.
Wahrscheinlich wars Zeile 274 im alten 289 im neuen.

Reply:
Okay, dann passt das so.